irmaclient/session.go

```diff
@@ -466,76 +479,63 @@ func (session *session) sendResponse(message interface{}) {
 			session.fail(&irma.SessionError{ErrorType: irma.ErrorSerialization, Info: "Type assertion failed"})
 			return
 		}
-
 		messageJson, err = json.Marshal(irmaSignature)
 		if err != nil {
 			session.fail(&irma.SessionError{ErrorType: irma.ErrorSerialization, Err: err})
 			return
 		}
-
-		if session.IsInteractive() {
-			var response disclosureResponse
-			if err = session.transport.Post("proofs", &response, irmaSignature); err != nil {
-				session.fail(err.(*irma.SessionError))
-				return
-			}
-			if response != "VALID" {
-				session.fail(&irma.SessionError{ErrorType: irma.ErrorRejected, Info: string(response)})
-				return
-			}
-		}
-		log, err = session.createLogEntry(message)
-		if err != nil {
-			irma.Logger.Warn(errors.WrapPrefix(err, "Failed to create log entry", 0).ErrorStack())
-			session.client.reportError(err)
-		}
+		ourResponse = irmaSignature
+		path = "proofs"
 	case irma.ActionDisclosing:
 		messageJson, err = json.Marshal(message)
 		if err != nil {
 			session.fail(&irma.SessionError{ErrorType: irma.ErrorSerialization, Err: err})
 			return
 		}
-		if session.IsInteractive() {
-			var response disclosureResponse
-			if err = session.transport.Post("proofs", &response, message); err != nil {
-				session.fail(err.(*irma.SessionError))
-				return
-			}
-			if response != "VALID" {
-				session.fail(&irma.SessionError{ErrorType: irma.ErrorRejected, Info: string(response)})
-				return
-			}
-		}
-		log, err = session.createLogEntry(message)
-		if err != nil {
-			irma.Logger.Warn(errors.WrapPrefix(err, "Failed to create log entry", 0).ErrorStack())
-			session.client.reportError(err)
-		}
+		ourResponse = message
+		path = "proofs"
 	case irma.ActionIssuing:
-		response := []*gabi.IssueSignatureMessage{}
-		if err = session.transport.Post("commitments", &response, message); err != nil {
+		ourResponse = message
+		path = "commitments"
+	}
+
+	if session.IsInteractive() {
+		if err = session.transport.Post(path, &serverResponse, ourResponse); err != nil {
```

Comment:
Not really, it is a by-effect of this refactor of the code. However, in practice it makes no difference because in case of `ActionIssuing`, the session is always interactive, i.e. `session.IsInteractive()` is never false. (Indeed, an issuance session can never be non-interactive: you need a remote server to receive the CL signatures from.)

Reply:
Maybe it is better then to move the `if session.Action == irma.ActionIssuing` from line 513 until 516 inside the `if session.IsInteractive()` block to make this explicit. Otherwise people might get triggered for potential nil pointer errors here.